grakn-engine/src/main/java/ai/grakn/engine/session/GraqlSession.java

```diff
@@ -218,7 +218,7 @@ void receiveQuery(Json json) {
                 String queryString = queryStringBuilder.toString();
                 queryStringBuilder = new StringBuilder();
 
-                queries = graph.graql().infer(infer).materialise(materialise).parseList(queryString);
+                queries = graph.graql().infer(infer).materialise(materialise).parseList(queryString).collect(toList());
```

Comment:
Actually I take it back - I'm not sure if this is possible: we do other stuff with the list of queries, such as killing them if they're compute queries, etc.